pkgs/development/python-modules/jsonschema/default.nix

```diff
@@ -1,28 +1,42 @@
-{ lib, buildPythonPackage, fetchPypi, python, isPy27
+{ lib
+, buildPythonPackage
+, fetchPypi
+, python
+, isPy27
 , attrs
-, functools32
 , importlib-metadata
-, mock
+, importlib-resources
 , nose
 , pyperf
 , pyrsistent
 , setuptools-scm
-, twisted
-, vcversioner
 }:
 
 buildPythonPackage rec {
   pname = "jsonschema";
-  version = "3.2.0";
+  version = "4.2.1";
+
+  dontPreferSetupPy = true;
+  format = "pyproject";
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "c8a85b28d377cc7737e46e2d9f2b4f44ee3c0e1deac6bf46ddefc7187d30797a";
+    sha256 = "sha256-OQcTRprmS4pYaYuzy8OFmr5pJbVlqXP4cyPvIbCaJ6g=";
   };
 
   nativeBuildInputs = [ setuptools-scm ];
-  propagatedBuildInputs = [ attrs importlib-metadata functools32 pyrsistent ];
-  checkInputs = [ nose mock pyperf twisted vcversioner ];
+
+  propagatedBuildInputs = [
+    attrs
+    importlib-metadata
+    importlib-resources
+    pyrsistent
+  ];
+
+  checkInputs = [
+    nose
+    pyperf
+  ];
 
   # zope namespace collides on py27
   doCheck = !isPy27;
```

Comment:
This should be removed and set `disabled = pythonOlder "3.7";` set. The min. [supported Python release is 3.7](https://github.com/Julian/jsonschema/blob/main/setup.cfg#L29).

Reply:
Done.